operators/pkg/instance-controller/status_inspection.go

```diff
@@ -15,6 +16,43 @@ import (
 	crownlabsv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
 )
 
+func (r *InstanceReconciler) getContainerStatus(ctx context.Context,
+	guiEnabled bool, service *v1.Service, ingress *networkingv1.Ingress,
+	instance *crownlabsv1alpha2.Instance, depl *appsv1.Deployment, startTimeVM time.Time) {
+	var ip string
+	url := ingress.GetAnnotations()["crownlabs.polito.it/probe-url"]
+
+	// iterate until the vm is running
+	for {
+		err := r.Client.Get(ctx, types.NamespacedName{
+			Namespace: depl.Namespace,
+			Name:      depl.Name,
+		}, depl)
+		if err == nil && depl.Status.ReadyReplicas > 0 {
+			break
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+
+	host := service.Name + "." + service.Namespace
+	port := "6080" // VNC
+	if !guiEnabled {
+		port = "22" // SSH
+	}
+
+	err := waitForConnection(host, port)
+	if err != nil {
+		klog.Error(fmt.Sprintf("Unable to check whether %v:%v is reachable", host, port))
+		klog.Error(err)
+	} else {
+		msg := "ContainerDeployment " + depl.Name + " in namespace " + depl.Namespace + " status update to READY."
+		r.setInstanceStatus(ctx, msg, "Normal", "ContainerReady", instance, ip, url)
```

Comment:
I suppose this should be VmiReady (although it is awkward), otherwise it is not recognized by the dashboard. 
These names will then need to be completely refactored.